crates/bevy_ecs/src/bundle.rs

```diff
@@ -203,19 +197,67 @@ use variadics_please::all_tuples;
 pub unsafe trait Bundle: DynamicBundle + Send + Sync + 'static {
     /// Gets this [`Bundle`]'s component ids, in the order of this bundle's [`Component`]s
     #[doc(hidden)]
-    fn component_ids(components: &mut ComponentsRegistrator, ids: &mut impl FnMut(ComponentId));
+    fn component_ids(
+        &self,
+        components: &mut ComponentsRegistrator,
+        ids: &mut impl FnMut(ComponentId),
+    );
 
     /// Gets this [`Bundle`]'s component ids. This will be [`None`] if the component has not been registered.
-    fn get_component_ids(components: &Components, ids: &mut impl FnMut(Option<ComponentId>));
+    fn get_component_ids(&self, components: &Components, ids: &mut impl FnMut(Option<ComponentId>));
 
     /// Registers components that are required by the components in this [`Bundle`].
     fn register_required_components(
+        &self,
         _components: &mut ComponentsRegistrator,
         _required_components: &mut RequiredComponents,
     );
 }
 
-/// Creates a [`Bundle`] by taking it from internal storage.
+/// Each bundle represents a static and fixed set of [`Component`] types.
+/// See the [`Bundle`] trait for a possibly dynamic set of [`Component`] types.
+///
+/// Implementers of the `Bundle` trait are called 'static bundles'.
+///
```

Comment:
A note somewhere around here about why we don't have a `StaticBundle: Bundle` trait bound would be helpful. I presume that's because it would break something (object-safety?), but it would be really helpful to explicitly record the reasoning.